src/App.js

```diff
@@ -30,24 +30,63 @@ const App = () => {
   // This starts state off as a 2D array of JS objects with
   // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
+  const [player, setPlayer] = useState(PLAYER_1);
+  const [winner, setWinner] = useState(null);
+  
+  const updateSquare = (updatedSquareId) => {
+  
+    const newSquares = [];
+    if (winner=== null) {
+    for (const row of squares) { 
+      const newRow = [];
+      newSquares.push(newRow);
+      for (const square of row) {
+        if (square.id === updatedSquareId && square.value === '') {
+          const updatedSquare = {
+            id: updatedSquareId,
+            value: player
+          }
+          newRow.push(updatedSquare);
+          const newPlayer = PLAYER_1 === player ? PLAYER_2 : PLAYER_1;
+          setPlayer(newPlayer);
 
-  // Wave 2
-  // You will need to create a method to change the square 
-  //   When it is clicked on.
-  //   Then pass it into the squares as a callback
-
-
-  const checkForWinner = () => {
-    // Complete in Wave 3
-    // You will need to:
-    // 1. Go accross each row to see if 
-    //    3 squares in the same row match
-    //    i.e. same value
-    // 2. Go down each column to see if
-    //    3 squares in each column match
-    // 3. Go across each diagonal to see if 
-    //    all three squares have the same value.
-
+        } else {
+          newRow.push(square);
+      
+        }
+      }
+    }
+  } else {
+    return;
+  }
+  checkForWinner(newSquares)
+  setSquares(newSquares)
+  }
+ 
+  const checkForWinner = (squares) => {
+    const winningCombos = [
```

Comment:
This is a clever approach. Consider how you could also loop over the 2D array to check for matches on rows and columns.